docs/architecture/psa-migration/dependencies-tls.md

```diff
@@ -0,0 +1,563 @@
+Dependencies of the TLS library on the Crypto library
```

Comment:
The point was mainly to support the analysis and make sure I didn't miss anything when creating tasks or looking for potential problems that would require further thought. I committed it because why not, but I agree it might be better to remove it now (it will still be in the git history if anyone cares).